File: typescript-in-action/demo/DemoApp/Models/UserBook.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace DemoApp.Models
{
    public class UserBook : Book
    {
        public DateTime Added { get; set; }

        public string[] Tags { get; set; }

        public string[] Lists { get; set; }
    }
}

File: typescript-in-action/demo/DemoApp/Scripts/app/UserBook.ts



import {Book} from './Book'
export class UserBook extends Book {

    
    public added: Date = null;
    public tags: string[] = [];
    public lists: string[] = [];

    public static fromJson (model: any): UserBook {
        var o = <UserBook>Book.fromJson(model);
        
        o.added = new Date(model.added);
        o.tags = model.tags;
        o.lists = model.lists;
        return o;
    }
}
